src/__tests__/components/fields/checkbox-group/checkbox-toggle-group.spec.tsx

```diff
@@ -0,0 +1,217 @@
+import { Button } from "@lifesg/react-design-system/button";
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+import { useState } from "react";
+import { ICheckboxGroupSchema } from "../../../../components/fields";
+import { FrontendEngine, IFrontendEngineData } from "../../../../components/frontend-engine";
+import {
+	ERROR_MESSAGE,
+	FRONTEND_ENGINE_ID,
+	TOverrideField,
+	TOverrideSchema,
+	getErrorMessage,
+	getSubmitButton,
+	getSubmitButtonProps,
+} from "../../../common";
+
+const SUBMIT_FN = jest.fn();
+const COMPONENT_ID = "field";
+const UI_TYPE = "checkbox";
+
+const renderComponent = (overrideField?: TOverrideField<ICheckboxGroupSchema>, overrideSchema?: TOverrideSchema) => {
+	const json: IFrontendEngineData = {
+		id: FRONTEND_ENGINE_ID,
+		sections: {
+			section: {
+				uiType: "section",
+				children: {
+					[COMPONENT_ID]: {
+						label: "Toggle",
+						uiType: UI_TYPE,
+						customOptions: {
+							styleType: "toggle",
+						},
+						options: [
+							{ label: "A", value: "Apple" },
+							{ label: "B", value: "Berry" },
+						],
+						...overrideField,
+					},
+					...getSubmitButtonProps(),
+				},
+			},
+		},
+		...overrideSchema,
+	};
+	return render(<FrontendEngine data={json} onSubmit={SUBMIT_FN} />);
+};
+
+const getToggles = (): HTMLElement[] => {
+	return screen.getAllByTestId("toggle-input");
```

Comment:
as we are using react-testing-library, we want to adhere to their guiding principles, querying by test-id is less preferred, instead try to query by role where possible.
reference: https://testing-library.com/docs/queries/about#priority

Reply:
Fixed